pyogrio/_io.pyx

```diff
@@ -2032,7 +2070,7 @@ def ogr_write(
                             field_value = str(field_value)
 
                         try:
-                            value_b = field_value.encode("UTF-8")
+                            value_b = field_value.encode(encoding)
```

Comment:
I am not fully sure this should be changed. It's an old RFC, but according to https://gdal.org/development/rfc/rfc23_ogr_unicode.html#oftstring-oftstringlist-fields field _values_ (so not field _names_) in the GDAL API's are always UTF8?

Reply:
It's indeed not 100% clear. Based on the next paragraph in that document, regarding the [olcstringsasutf8-capability-flag](https://gdal.org/development/rfc/rfc23_ogr_unicode.html#olcstringsasutf8-capability-flag), I interprete that for drivers/layers that return True for this flag, the field values should always be in UTF-8 (for most drivers), and for other drivers it should be in the encoding you want it to be written (e.g. for .csv). This is how I intended it to work.

What do you think?

Reply:
> for drivers/layers that return True for this flag, the field values should always be in UTF-8 (for most drivers), and for other drivers it should be in the encoding you want it to be written

My interpretation of the docs is in line with this.

Reply:
Indeed, that sounds correct